File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations): 
    citations.sort(reverse=True)
    for h in range(citations[0], 0, -1):
        up = 0
        down = 0
        for citation in citations:
            if citation >= h:
                up += 1
            else:
                down += 1
        if up >= h and down <= h:
            return h
    return 0 # [0, 0, 0, 0, 0, 0] 0

File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        Arrays.sort(citations);
        for (int h = citations[citations.length - 1]; h >= 0; h--) {
            int cnt = 0;
            for (int citation : citations) {
                if (citation >= h) cnt++;
            }
            if (cnt >= h) return h;
        }
        int answer = 0;
        return answer;
    }
}